fastlane/lanes/localization.rb

```diff
@@ -165,71 +165,86 @@
+  # @param [Boolean] skip_git_push If true, will skip the `git push` at the end. Default to false. Inferred to `true` if `skip_commit` is `true`.
   #
-  desc 'Downloads translated metadata from GlotPress'
-  lane :download_metadata_strings do |options|
-    skip_release_notes = options.fetch(:skip_release_notes, false)
-    version = skip_release_notes ? nil : options.fetch(:version, current_release_version)
-
-    skip_commit = options.fetch(:skip_commit, false)
-    skip_git_push = options.fetch(:skip_git_push, false)
+  # @return [void]
+  #
+  lane :download_metadata_strings do |app: nil, version: current_release_version, skip_release_notes: false, skip_commit: false, skip_git_push: false|
+    version = nil if skip_release_notes
 
     # If no `app:` is specified, call this for both WordPress and Jetpack
-    app_param = options[:app]
-    apps = if app_param.nil?
+    apps = if app.nil?
              %i[wordpress jetpack]
            else
-             Array(app_param.to_s.downcase.to_sym)
+             Array(app.to_s.downcase.to_sym)
            end
 
-    apps.each do |app|
-      app_values = APP_SPECIFIC_VALUES[app]
+    apps.each do |current_app|
+      app_values = APP_SPECIFIC_VALUES[current_app]
       metadata_source_dir = File.join(PROJECT_ROOT_FOLDER, 'WordPress', app_values[:metadata_dir])
+      metadata_download_path = File.join(FASTLANE_FOLDER, app_values[:metadata_dir], 'android')
+      locales = { wordpress: WP_RELEASE_NOTES_LOCALES, jetpack: JP_RELEASE_NOTES_LOCALES }[current_app]
 
       files = {
         play_store_app_title: { desc: 'title.txt', max_size: 30 },
         play_store_promo: { desc: 'short_description.txt', max_size: 80 },
         play_store_desc: { desc: 'full_description.txt', max_size: 4000 }
       }
+
       unless skip_release_notes
         version_suffix = version.split('.').join
-        files["release_note_#{version_suffix}"] = { desc: 'changelogs/default.txt', max_size: 500, alternate_key: "release_note_short_#{version_suffix}" }
+        files["release_note_#{version_suffix}"] = {
+          desc: 'changelogs/default.txt',
+          max_size: 500,
+          alternate_key: "release_note_short_#{version_suffix}"
+        }
+
+        # Clear release notes if the source file is empty
+        source_notes_file = File.join(metadata_source_dir, 'release_notes.txt')
+        if !File.exist?(source_notes_file) || File.read(source_notes_file).strip.empty?
+          all_app_locales = locales + [['en', 'en-US']]
+          all_app_locales.each do |_, google_play_locale|
+            changelog_dir = File.join(metadata_download_path, google_play_locale, 'changelogs')
+            FileUtils.mkdir_p(changelog_dir)
+            File.write(File.join(changelog_dir, 'default.txt'), '')
+          end
+          # Skip downloading release notes translations since the source is empty
+          files.delete("release_note_#{version_suffix}")
```

Comment:
Reading this made me think at first that this was deleting a file with this name. Took me a minute to realize this was instead just removing the entry with that key from the `files` `Hash` instead.

Wouldn't it be more readable to do the check first, and only assign the value of `files` for that key (on line 201) if not empty?

In other words, write this block of code as:
```ruby
if !File.exist?(…) || File.read(…).strip.empty?
  # Clear release notes if the source file is empty
  # …
else
  # Make sure to also download the release notes translations (given we had non-empty original copy)
  files["release_note_#{version_suffix}"] = { … }
end
```